### PR TITLE
Update CRD spec to include list of repositories

### PR DESCRIPTION
**Pull request by andresmgot**
### Description of the change

As explained in the design document, the option chosen to implement OCI registries requires a small modification of the AppRepository CRD to include the list of repositories.

### Applicable issues

  - ref #2232

### CODE REVIEW

**Review by antgamdia — APPROVED** [review]
Great! So, this is just a list of repository names for CRD to have a kind of reference/index with the repos that it contains, I guess

**Comment by andresmgot** [comment]
> Great! So, this is just a list of repository names for CRD to have a kind of reference/index with the repos that it contains, I guess

Yes, since OCI registries don't have an index endpoint, in this case, it's manual.

File: cmd/apprepository-controller/pkg/apis/apprepository/v1alpha1/types.go [hunk 1]
```diff
@@ -45,6 +45,9 @@ type AppRepositorySpec struct {
 	// in the same namespace as the AppRepository and should be included
 	// automatically for matching images.
 	DockerRegistrySecrets []string `json:"dockerRegistrySecrets,omitempty"`
+	// In case of an OCI type, the list of repositories is needed
+	// as there is no API for the index
+	Repositories []string `json:"repositories,omitempty"`
```
Comment (absoludity) [inline cmd/apprepository-controller/pkg/apis/apprepository/v1alpha1/types.go:50]: Just a thought, feel free to ignore: This object to which you're adding the `Repositories` is the spec of an `AppRepository`. Generally, an `AppRepository` as a repository of apps (whether that's a helm repository or something else in the future, such as, in this case, an OCI registry), so it initially hits some dissonance to then see a field called `Repositories`. Might be clearer to name it `OCIRepositories` so `appRepository.Spec.Repositories` becomes `appRepository.Spec.OCIRepositories`. See what you think.

Comment (andresmgot) [reply cmd/apprepository-controller/pkg/apis/apprepository/v1alpha1/types.go:50]: makes sense to me

**Pull request merged**